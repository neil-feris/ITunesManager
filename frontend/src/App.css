@import url("https://fonts.googleapis.com/css2?family=Lora:wght@400;700&family=Montserrat+Alternates:wght@400;700&display=swap");

.MuiTypography-h4 {
  font-family: "Lora", serif !important;
  text-align: center;
}
