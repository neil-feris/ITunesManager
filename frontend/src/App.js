import CssBaseline from "@mui/material/CssBaseline";
import "./App.css";

import React from "react";
import { Routes, Route } from "react-router-dom";
import Header from "./components/Header";
import Main from "./components/Main";
import Favourites from "./components/Favourites";

function App() {
  return (
    <>
      <CssBaseline />
      <Header />
      <Routes>
        <Route path="/" element={<Main />} />
        <Route path="/favourites" element={<Favourites />} />
      </Routes>
    </>
  );
}

export default App;
