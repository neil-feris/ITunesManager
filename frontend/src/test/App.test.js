// import React
import React from "react";

// import TestRenderer for snapshot testing
import TestRenderer from "react-test-renderer";

// import App
import App from "../App";

// snapshot test
it("renders App correctly", () => {
  const tree = TestRenderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
