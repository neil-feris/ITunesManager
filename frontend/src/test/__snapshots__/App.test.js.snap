// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-hip9hq-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <div
        className="MuiTypography-root MuiTypography-h5 css-73zt9z-MuiTypography-root"
      >
        <a
          className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1n8yj6x-MuiTypography-root-MuiLink-root"
          href="/"
          onBlur={[Function]}
          onFocus={[Function]}
        >
          Media Favourites Manager
        </a>
      </div>
      <a
        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1atexe4-MuiTypography-root-MuiLink-root"
        href="/favourites"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Favourites
      </a>
    </div>
  </header>,
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiBox-root css-r3ax1p"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1jvq7n4-MuiTypography-root"
          >
            Search iTunes
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  css-xunhrp-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            onClick={[Function]}
          >
            <select
              className="MuiNativeSelect-select MuiNativeSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-ffg8md-MuiNativeSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              defaultValue="music"
              id="mediaType"
              multiple={false}
              name="mediaType"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
            >
              <option
                value={
                  Array [
                    "movie",
                  ]
                }
              >
                Movie
              </option>
              <option
                value={
                  Array [
                    "podcast",
                  ]
                }
              >
                Podcast
              </option>
              <option
                value={
                  Array [
                    "music",
                  ]
                }
              >
                Music
              </option>
              <option
                value={
                  Array [
                    "musicVideo",
                  ]
                }
              >
                Music Video
              </option>
              <option
                value={
                  Array [
                    "audiobook",
                  ]
                }
              >
                Audiobook
              </option>
              <option
                value={
                  Array [
                    "shortFilm",
                  ]
                }
              >
                Short Film
              </option>
              <option
                value={
                  Array [
                    "tvShow",
                  ]
                }
              >
                TV Show
              </option>
              <option
                value={
                  Array [
                    "software",
                  ]
                }
              >
                Software
              </option>
              <option
                value={
                  Array [
                    "ebook",
                  ]
                }
              >
                Ebook
              </option>
              <option
                value={
                  Array [
                    "all",
                  ]
                }
              >
                All
              </option>
            </select>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-ihdtdm"
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary  css-xunhrp-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
            onClick={[Function]}
          >
            <select
              className="MuiNativeSelect-select MuiNativeSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-ffg8md-MuiNativeSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
              id="searchType"
              multiple={false}
              name="searchType"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
            >
              <option
                value={
                  Array [
                    "musicArtist",
                  ]
                }
              >
                Artist
              </option>
              <option
                value={
                  Array [
                    "musicTrack",
                  ]
                }
              >
                Music Track
              </option>
              <option
                value={
                  Array [
                    "album",
                  ]
                }
              >
                Album
              </option>
              <option
                value={
                  Array [
                    "musicVideo",
                  ]
                }
              >
                Music Video
              </option>
              <option
                value={
                  Array [
                    "mix",
                  ]
                }
              >
                Mix
              </option>
              <option
                value={
                  Array [
                    "song",
                  ]
                }
              >
                Song
              </option>
            </select>
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
              data-testid="ArrowDropDownIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
            <fieldset
              aria-hidden={true}
              className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="css-ihdtdm"
              >
                <span
                  className="notranslate"
                >
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
          <div
            className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary css-14gf15v-MuiInputBase-root-MuiInput-root"
            onClick={[Function]}
            onKeyPress={[Function]}
          >
            <input
              className="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search"
              type="text"
            />
          </div>
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-s3awaj-MuiButtonBase-root-MuiButton-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            Search
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
